### PR TITLE
Update/use system absolute path not classpath input file

### PR DESCRIPTION
**Pull request by dinilatgit**

### CODE REVIEW

File: metrics/src/main/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReader.java [hunk 1]
```diff
@@ -30,8 +32,9 @@
 
     public static List<MetricFunctionsSpec> readFromInputFile(String InputFilename) {
         List<MetricFunctionsSpec> metricFunctionSpecList = new ArrayList<>();
-        InputStream inputStream = MetricFunctionsReader.class.getResourceAsStream(InputFilename);
+        String inputFile = new File(InputFilename).toString();
         try {
+            InputStream inputStream = new FileInputStream(inputFile);
```
Comment (hridyeshpant) [inline metrics/src/main/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReader.java]: can't we use InputFilename parameter directly in new FileInputStream(). e.g new FileInputStream(InputFilename);

File: metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/service/graphite/GraphiteQueryServiceTest.java [hunk 2]
```diff
@@ -60,8 +58,9 @@ public void initMetricSourceSinkConfigSetup() throws Exception {
         MockitoAnnotations.initMocks(this);
         val config = new TypesafeConfigLoader(APP_ID).loadMergedConfig();
         metricSourceSinkConfig = config.getConfig(METRIC_SOURCE_SINK);
-        val functionsInputFile = "/config/functions-test.txt";
-        metricFunctionsSpec = MetricFunctionsReader.readFromInputFile(functionsInputFile).get(0);
+        val functionsInputFile = "config/functions-test.txt";
+        metricFunctionsSpec = MetricFunctionsReader.readFromInputFile(
+                ClassLoader.getSystemResource(functionsInputFile).getPath()).get(0);
```
Comment (hridyeshpant) [inline metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/service/graphite/GraphiteQueryServiceTest.java]: don't understand this. you are passing ClassLoader in the test but in actual you are using 
```val input_file = INPUT_FILE_PATH + INPUT_FUNCTIONS_FILENAME;
MetricFunctionsReader.readFromInputFile(input_file)```

File: metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReaderTest.java [hunk 3]
```diff
@@ -15,8 +15,9 @@
 
     @Test
     public void testReadFromInputFile(){
-        val functionInputFileName = "/config/functions-test.txt";
-        List<MetricFunctionsSpec> metricFunctionsSpecList = MetricFunctionsReader.readFromInputFile(functionInputFileName);
+        val functionInputFileName = "config/functions-test.txt";
+        List<MetricFunctionsSpec> metricFunctionsSpecList = MetricFunctionsReader.readFromInputFile
+                (ClassLoader.getSystemResource(functionInputFileName).getPath());
```
Comment (hridyeshpant) [inline metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReaderTest.java]: same as above

File: metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/graphite/ConstructSourceURITest.java [hunk 4]
```diff
@@ -6,15 +6,17 @@
...
 public class ConstructSourceURITest {
 
     @Test
     public void testGetGraphiteURI() {
-        val functionsInputFileName = "/config/functions-test.txt";
+        val functionsInputFileName = "config/functions-test.txt";
         val uri = "samplegraphitehosturi/render?until=now&format=json&target=sumSeries(a.b.c)&from=-30s";
-        MetricFunctionsSpec metricFunctionsSpec = MetricFunctionsReader.readFromInputFile(functionsInputFileName).get(0);
+        MetricFunctionsSpec metricFunctionsSpec = MetricFunctionsReader.
+                readFromInputFile(ClassLoader.getSystemResource(functionsInputFileName).getPath()).get(0);
```
Comment (hridyeshpant) [inline metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/graphite/ConstructSourceURITest.java]: same as above

File: metrics/src/main/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReader.java (hunk 1 above)
Comment (dinilatgit) [reply metrics/src/main/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReader.java]: Fixed.

File: metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/service/graphite/GraphiteQueryServiceTest.java (hunk 2 above)
Comment (dinilatgit) [reply metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/service/graphite/GraphiteQueryServiceTest.java]: Deployment uses absolute path of file - "/config/functions.txt". 
Updated test cases to reflect the same as well.

File: metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReaderTest.java (hunk 3 above)
Comment (dinilatgit) [reply metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/MetricFunctionsReaderTest.java]: Deployment uses absolute path of file - "/config/functions.txt". 
Updated test cases to reflect the same as well.

File: metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/graphite/ConstructSourceURITest.java (hunk 4 above)
Comment (dinilatgit) [reply metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/source/graphite/ConstructSourceURITest.java]: Deployment uses absolute path of file - "/config/functions.txt". 
Updated test cases to reflect the same as well.

**Pull request merged**